packages/atoms/src/Modal/ModalBody.tsx

```diff
@@ -0,0 +1,57 @@
+import { TouchableOpacity } from 'react-native';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import React, { ReactNode } from 'react';
+import { Box } from '../Box';
+import { Icon } from '../Icon/Icon';
+import { Text } from '../Text';
+
+type Props = {
+    onClose: () => void;
+    children: ReactNode;
+    title: string;
+    hasBackArrow: boolean;
+    onBackArrowClick: () => void;
+};
+
+const modalWrapperStyle = prepareNativeStyle(utils => ({
+    backgroundColor: utils.colors.gray100,
+    borderTopLeftRadius: utils.borders.radii.xLarge,
+    borderTopRightRadius: utils.borders.radii.xLarge,
+}));
+
+const CLOSE_BUTTON_SIZE = 40;
+const closeButtonStyle = prepareNativeStyle(utils => ({
+    backgroundColor: utils.colors.gray100,
+    borderRadius: utils.borders.radii.round,
+    height: CLOSE_BUTTON_SIZE,
+    width: CLOSE_BUTTON_SIZE,
+    justifyContent: 'center',
+    alignItems: 'center',
+}));
+
+export const ModalBody = ({ onClose, title, hasBackArrow, onBackArrowClick, children }: Props) => {
+    const { applyStyle } = useNativeStyles();
+
+    return (
+        <Box style={applyStyle(modalWrapperStyle)}>
+            <Box
+                flexDirection="row"
+                justifyContent="space-between"
+                alignItems="center"
+                paddingVertical="md"
+                paddingHorizontal="md"
+            >
+                {hasBackArrow && (
+                    <TouchableOpacity onPress={onBackArrowClick}>
+                        <Icon type="chevronLeft" />
+                    </TouchableOpacity>
+                )}
+                <Text variant="titleSmall">{title}</Text>
+                <TouchableOpacity onPress={onClose} style={applyStyle(closeButtonStyle)}>
+                    <Icon type="close" />
+                </TouchableOpacity>
+            </Box>
```

Comment:
Probably this will be deleted along with the styles to reuse the Icon in ListItem PR